services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/OpenHouseInternalRepositoryImpl.java

```diff
@@ -434,75 +454,137 @@ private boolean checkIfTableTypeAdded(
-    return catalog.tableExists(
-        TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()));
+    long startTime = System.currentTimeMillis();
+    TableIdentifier tableId =
+        TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId());
+    boolean result = catalog.tableExists(tableId);
+    long executionTime = System.currentTimeMillis() - startTime;
+    meterRegistry
+        .timer(MetricsConstant.REPO_TABLE_EXISTS_TIME)
+        .record(executionTime, TimeUnit.MILLISECONDS);
+    log.info("existsById for table {} took {} ms", tableId, executionTime);
+    return result;
   }
 
   @Override
   public void deleteById(TableDtoPrimaryKey tableDtoPrimaryKey) {
-    catalog.dropTable(
-        TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()),
-        true);
+    long startTime = System.currentTimeMillis();
+    TableIdentifier tableId =
+        TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId());
+    catalog.dropTable(tableId, true);
+    long executionTime = System.currentTimeMillis() - startTime;
+    meterRegistry
+        .timer(MetricsConstant.REPO_TABLE_DELETE_TIME)
+        .record(executionTime, TimeUnit.MILLISECONDS);
+    log.info("deleteById for table {} took {} ms", tableId, executionTime);
   }
 
   @Override
   public List<TableDto> findAllByDatabaseId(String databaseId) {
+    long startTime = System.currentTimeMillis();
     List<Table> tables =
         catalog.listTables(Namespace.of(databaseId)).stream()
             .map(tableIdentifier -> catalog.loadTable(tableIdentifier))
             .collect(Collectors.toList());
-    return tables.stream()
-        .map(
-            table ->
-                convertToTableDto(
-                    table, fsStorageProvider, partitionSpecMapper, policiesMapper, tableTypeMapper))
-        .collect(Collectors.toList());
+    List<TableDto> result =
+        tables.stream()
+            .map(
+                table ->
+                    convertToTableDto(
+                        table,
+                        fsStorageProvider,
+                        partitionSpecMapper,
+                        policiesMapper,
+                        tableTypeMapper))
+            .collect(Collectors.toList());
+    long executionTime = System.currentTimeMillis() - startTime;
+    meterRegistry
+        .timer(MetricsConstant.REPO_TABLES_FIND_BY_DATABASE_TIME)
+        .record(executionTime, TimeUnit.MILLISECONDS);
+    log.info("findAllByDatabaseId for database {} took {} ms", databaseId, executionTime);
+    return result;
   }
 
   @Override
   public List<TableDto> searchTables(String databaseId) {
-    return catalog.listTables(Namespace.of(databaseId)).stream()
-        .map(tablesIdentifier -> mapper.toTableDto(tablesIdentifier))
-        .collect(Collectors.toList());
+    long startTime = System.currentTimeMillis();
```

Comment:
is it possible to introduce an annotation that can apply to multiple methods? will significantly help with code readability.

`@Timed(metric=MetricsConstant.REPO_TABLES_SEARCH_BY_DATABASE_TIME, unit=MILLISECONDS)`


For methods with internal code to be measured, we could keep the current state.

Reply:
I reverted this https://github.com/linkedin/openhouse/pull/62/commits/85d31981b855cfdcd0979d9bceab44f605a6984d because it has become even more verbose when I added logging

Reply:
Loved the commit! we don't need verbose logs. A templatized log should be enough, for example:

```
time taken for <method signature> was X Seconds
```

example: `time taken for save was 20 seconds`

^ could we still apply that lovely commit?

Reply:
I wanted table id to be part of the log, it's meaningful and helps to correlate logs, would you still suggest defining aspect that accesses method arguments? We'd need an aspect method for each signature

Reply:
no, how about this:
```
@Timed(metric=MetricsConstant.REPO_TABLES_SEARCH_BY_DATABASE_TIME, unit=MILLISECONDS, contextLog="tablename: dbX.tableY")
```

Reply:
Thanks @HotSushi , refactored into aspect